watertap/unit_models/ion_exchange_0D.py

```diff
@@ -0,0 +1,1352 @@
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "target_ion",
+        ConfigValue(default="Ca_2+", domain=str, description="Target Ion"),
+    )
+
+    CONFIG.declare(
+        "regenerant",
+        ConfigValue(default="HCl", domain=str, description="Regenerant"),
+    )
+
+    def build(self):
+        super().build()
+
+        ### REFERENCES ###
+        # LeVan, M. D., Carta, G., & Yon, C. M. (2019).
+        # Section 16: Adsorption and Ion Exchange.
+        # Perry's Chemical Engineers' Handbook, 9th Edition.
+
+        # Crittenden, J. C., Trussell, R. R., Hand, D. W., Howe, K. J., & Tchobanoglous, G. (2012).
+        # Chapter 16: Ion Exchange.
+        # MWH's Water Treatment (pp. 1263-1334): John Wiley & Sons, Inc.
+
+        # DOWEX Ion Exchange Resins Water Conditioning Manual
+        # https://www.lenntech.com/Data-sheets/Dowex-Ion-Exchange-Resins-Water-Conditioning-Manual-L.pdf
+
+        # Inamuddin, & Luqman, M. (2012).
+        # Ion Exchange Technology I: Theory and Materials.
+
+        ion_set = self.config.property_package.ion_set
+        target_ion = self.config.target_ion
+
+        # this creates blank scaling factors, which are populated later
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        self.gravity = Param(
+            initialize=9.81, units=pyunits.m / pyunits.s**2, doc="Gravity [m/s2]"
+        )
+
+        self.pi = Param(initialize=3.14159, units=pyunits.dimensionless, doc="Pi")
+
+        self.diff_ion_resin = Param(
+            initialize=1e-13,
+            units=pyunits.m**2 / pyunits.s,
+            doc="Diffusivity of ion resin [m2/s]",  # Perry's
+        )
+
+        self.underdrain_h = Param(
+            initialize=0.5, units=pyunits.m, doc="Underdrain height [m]"  # Perry's
+        )
+
+        self.distributor_h = Param(
+            initialize=1.5, units=pyunits.m, doc="Distributor height [m]"  # Perry's
+        )
+
+        self.holdup_A = Param(
+            initialize=21, units=pyunits.dimensionless, doc="Holdup eq A parameter"
+        )
+
+        self.holdup_B = Param(
+            initialize=99.72, units=pyunits.dimensionless, doc="Holdup eq B parameter"
+        )
+
+        self.holdup_exp = Param(
+            initialize=0.28, units=pyunits.dimensionless, doc="Holdup eq exponent"
+        )
+
+        self.Pe_p_A = Param(
+            initialize=0.05,
+            units=pyunits.dimensionless,
+            doc="Pe particle eq A parameter",
+        )
+
+        self.Pe_p_exp = Param(
+            initialize=0.48, units=pyunits.dimensionless, doc="Pe particle eq exp"
+        )
+
+        self.Sh_A = Param(
+            initialize=1.09, units=pyunits.dimensionless, doc="Sh eq A parameter"
+        )
```

Comment:
A lot of these parameter names are unclear and the `doc` strings added don't clear things up. If these come from one of the references, it may be good to put in a note about that.